public/app/plugins/panel/heatmap/heatmap_ctrl.ts

```diff
@@ -321,31 +330,53 @@ export class HeatmapCtrl extends MetricsPanelCtrl {
     this.render();
   }
 
-  seriesHandler(dataFrame: any) {
-    if (dataFrame.datapoints === undefined) {
-      throw new Error('Heatmap error: data should be a time series');
-    }
+  framesToTimeSeries(dataFrames: DataFrame[]): TimeSeries[] {
```

Comment:
This looks very similar to https://github.com/grafana/grafana/blob/a3dd67ab6da592319adeb50a0e7a0df268c94d2a/public/app/plugins/panel/graph/data_processor.ts#L15
I think it would be nice to share this function between panels.